notes/typed-js.md

```diff
@@ -9,3 +11,37 @@ https://doc.rust-lang.org/nomicon/other-reprs.html
 #[repr(transparent)]
 struct Array(JSAny);
 ```
+
+## Modifying JSAny
+
+```rust
+trait TsType {}
+
+struct Unknown();
+impl TsType for Unknown {}
+
+struct JSAny<D: Dealloc, T: TsType = Unknown> {
+   ...,
+   _p: PhantomData<T>,
+}
+
+struct Array<T: TsType>();
+
+impl<D: Dealloc, T: TsType> JSAny<D, Array<T>> {
+   fn at(self, i: i32) -> JSAny<D, T>;
+}
+```
+
+## Combining
+
+```rust
```

Comment:
I am trying to catch this idea - TSAny "lifts" JSAny somehow, but, I'd appreciate if you add a phrase here describing TSAny vs JSAny difference in few words, please!